src/test/java/com/puppycrawl/tools/checkstyle/api/DetailASTTest.java

```diff
@@ -117,10 +124,94 @@ public void testInsertSiblingBetween() throws Exception {
     }
 
     @Test
+    public void testClearBranchTokenTypes() throws Exception {
+        final DetailAST parent = new DetailAST();
+        final DetailAST child = new DetailAST();
+        parent.setFirstChild(child);
+
+        final List<Consumer<DetailAST>> clearBranchTokenTypesMethods = Arrays.asList(
+            ast -> child.setFirstChild(ast),
+            ast -> child.setNextSibling(ast),
+            ast -> child.addPreviousSibling(ast),
+            ast -> child.addNextSibling(ast),
+            ast -> child.addChild(ast),
+            ast -> {
+                try {
+                    Whitebox.invokeMethod(child, "setParent", ast);
+                }
+                // -@cs[IllegalCatch] Cannot avoid catching it.
+                catch (Exception exception) {
+                    throw new IllegalStateException(exception);
+                }
+            }
+        );
+
+        for (Consumer<DetailAST> method : clearBranchTokenTypesMethods) {
+            final BitSet branchTokenTypes = Whitebox.invokeMethod(parent, "getBranchTokenTypes");
+            method.accept(null);
+            final BitSet branchTokenTypes2 = Whitebox.invokeMethod(parent, "getBranchTokenTypes");
+            assertNotSame(branchTokenTypes, branchTokenTypes2);
```

Comment:
Can we then assert the resulting values are the same?
`NotSame` atleast verifies instance is different.